descriptor/qsarmolecular/src/main/java/org/openscience/cdk/qsar/descriptors/molecular/AcidicGroupCountDescriptor.java

```diff
@@ -114,6 +114,8 @@ public DescriptorValue calculate(IAtomContainer atomContainer) {
             throw new IllegalStateException("descriptor is not initalised, invoke 'initalise' first");
         }
 
+        atomContainer = clone(atomContainer); // don't mod original
```

Comment:
Yes, very much agreed, this is not how it should be. In the past I have changed code to not have to do this. But as a workaround this will do for now.

Reply:
(But that explains why you will find that flag copying and restoring code around :)